hibernate-core/src/main/java/org/hibernate/internal/NaturalIdMultiLoadAccessStandard.java

```diff
@@ -132,6 +138,53 @@ public List<T> multiLoad(Object... ids) {
 
 	}
 
+	protected void performAnyNeededCrossReferenceSynchronizations() {
+		final NaturalIdMapping naturalIdMapping = entityDescriptor.getNaturalIdMapping();
+
+		if ( !naturalIdMapping.isMutable() ) {
+			// only mutable natural-ids need this processing
+			return;
+		}
+
+		if ( ! session.isTransactionInProgress() ) {
+			// not in a transaction so skip synchronization
+			return;
+		}
+
+		final PersistenceContext persistenceContext = session.getPersistenceContextInternal();
+		final Collection<?> cachedPkResolutions =
+				persistenceContext.getNaturalIdResolutions()
+						.getCachedPkResolutions( entityDescriptor );
+		final boolean loggerDebugEnabled = LoaderLogging.LOADER_LOGGER.isDebugEnabled();
+		for ( Object pk : cachedPkResolutions ) {
+			final EntityKey entityKey = session.generateEntityKey( pk, entityDescriptor );
+			final Object entity = persistenceContext.getEntity( entityKey );
+			final EntityEntry entry = persistenceContext.getEntry( entity );
+
+			if ( entry == null ) {
+				if ( loggerDebugEnabled ) {
+					LoaderLogging.LOADER_LOGGER.debugf(
+							"Cached natural-id/pk resolution linked to null EntityEntry in persistence context : %s#%s",
+							entityDescriptor.getEntityName(),
+							pk
+					);
+				}
+				continue;
+			}
+
+			if ( !entry.requiresDirtyCheck( entity ) ) {
+				continue;
+			}
+
+			// MANAGED is the only status we care about here...
+			if ( entry.getStatus() != Status.MANAGED ) {
+				continue;
+			}
+
+			persistenceContext.getNaturalIdResolutions().handleSynchronization( pk, entity, entityDescriptor );
+		}
+	}
```

Comment:
Is this just a copy/paste job from `BaseNaturalIdLoadAccessImpl`?

Reply:
98% the same yes

Reply:
Well can we extract it out then, please?

Reply:
> Well can we extract it out then, please?

How could I possibly refuse, now that you asked so nicely :D